leakcanary-object-watcher-android/src/main/java/leakcanary/ServiceWatcher.kt

```diff
@@ -54,6 +54,10 @@ class ServiceWatcher(private val reachabilityWatcher: ReachabilityWatcher) : Ins
           }
         }
         Handler.Callback { msg ->
+          if (msg.obj !is IBinder) {
+            return@Callback false
+          }
```

Comment:
On the bad device, once you add that fix, do we have other messages where msg.obj is a IBinder and can we still do service death detection? If yes then this fix is great, if no then I wonder if we could detect this ahead of time and not set up the hook at all. Thoughts?

Reply:
So far I haven't noticed messages beyond the `IBinder` or `ActivityClientRecord` types. I'm not sure what other type of detection we're looking for.

Reply:
I guess I just want to confirm you're still also seeing IBinder messages on that device.